pkg/operator/vspherecontroller/vspherecontroller.go

```diff
@@ -717,3 +733,95 @@ func (c *VSphereController) createStorageClassController() storageclasscontrolle
 	)
 	return storageClassController
 }
+
+func getUserAndPassword(namespace string, secretName string, infra *ocpv1.Infrastructure, configMapLister corelister.ConfigMapLister, secretInformer corev1informers.SecretInformer,
+) (string, string, error) {
+	secret, err := secretInformer.Lister().Secrets(namespace).Get(secretName)
+	if err != nil {
+		return "", "", err
+	}
+
+	vCenterName, err := getvCenterName(infra, configMapLister)
+	if err != nil {
+		return "", "", err
+	}
+
+	// CCO generates a secret that contains dynamic keys, for example:
+	// oc get secret/vmware-vsphere-cloud-credentials -o json | jq .data
+	// {
+	//   "vcenter.xyz.vmwarevmc.com.password": "***",
+	//   "vcenter.xyz.vmwarevmc.com.username": "***"
+	// }
+	// So we need to figure those keys out
+	var usernameKey, passwordKey string
+
+	if len(secret.Data) > 2 {
+		klog.Warningf("CSI driver can only connect to one vcenter, more than 1 set of credentials found for CSI driver")
+	}
+
+	usernameKey = vCenterName + ".username"
+	passwordKey = vCenterName + ".password"
+
+	if usernameKey == "" || passwordKey == "" {
+		return "", "", fmt.Errorf("could not find vSphere credentials in secret %s/%s", secret.Namespace, secret.Name)
+	}
+
+	// Get username and pass from secret created by CCO
+	username := string(secret.Data[usernameKey])
+	password := string(secret.Data[passwordKey])
+
+	// The CSI driver expects a password with any quotation marks and backslashes escaped.
+	// xref: https://github.com/kubernetes-sigs/vsphere-csi-driver/issues/121
+	password = reEscape.ReplaceAllString(password, `\$0`)
+	return username, escapeQuotesAndBackslashes(password), nil
```

Comment:
Are we escaping this two times? didn't we just escape this above?

Reply:
Good catch; this mistake is a result of a code refactoring where I moved the logic to `escapeQuotesAndBackslashes()` in order to unit-test it. PTAL